README.md

```diff
@@ -1,24 +1,272 @@
-# Requirements
+Corva python-sdk is a framework for building stream, scheduled and task apps.
+
+## Contents
+
+- [Requirements](#requirements)
+- [Installation](#installation)
+- [App types](#app-types)
+- [Event](#event)
+- [Api](#api)
+- [Cache](#cache)
+- [Contributing](#contributing)
+
+## Requirements
+
 Python 3.8+
 
-# Local development
+- [Pydantic](https://github.com/samuelcolvin/pydantic)
+- [Redis](https://pypi.org/project/redis/)
+- [Requests](https://pypi.org/project/requests/)
+- [urllib3](https://pypi.org/project/urllib3/)
+
+## Installation
+
+```console
+$ pip install corva-sdk
+```
+
+## App types
+
+There are three app types, that you can build: `stream`, `scheduled` and `task`.<br>
+**Note**: it is recommended to use type hints like in examples below,
+so that editors and tools can give you better support.
+
+#### Stream
+
+```python
+from corva import Api, Cache, Corva, StreamEvent
+
+
+# 1 define a function with required parameters, that will be provided by sdk
+def stream_app(event: StreamEvent, api: Api, cache: Cache):
+    """Main logic function"""
+    pass
+
+
+# 2 define a function that will be run by AWS lambda
+def lambda_handler(event, context):
+    """AWS lambda handler"""
+    corva = Corva()  # 3 initialize Corva
+    corva.stream(stream_app, event)  # 4 run stream_app by passing it and event to Corva.stream
+```
+
+`Corva.stream` provides two additional parameters:
+
+- `filter_by_timestamp` - if set to `True` will take the latest processed
+  `timestamp` from cache and filter out records from event with 
+  either smaller or same `timestamp`;
+- `filter_by_depth` - if set to `True` will take the latest processed
+  `measured_depth` from cache and filter out records from event with
+  either smaller or same `measured_depth`.
```

Comment:
as a user who faces this library for the first time - it's not clear to me what is it "Event", `measured_depth`

Reply:
added link to event section. we need to add a section describing event fields, but this should be done after deleting obsolete event fields. as for now lets leave it as it is